en/docs/learn/jarm-for-oauth2.md

```diff
@@ -0,0 +1,162 @@
+            In the default authorization flow (for default response modes), the authorization response parameters, including the authorization code and ID token, are sent as human-readable plain text to the redirect URL, as shown below.</br></br>
+            <code>https://&lt;CLIENT_HOST&gt;/redirects/redirect1?code=&lt;AUTH_CODE&gt;&session_state=&lt;SESSION_STATE&gt;</code></br></br>
+            To change this response to a more secure JWT format, you need to use JARM.
+        </td>
+    </tr>
+</table>
+
+## Using JARM
+
+When [JARM is enabled in WSO2 Identity Server](#enable-jarm), clients are able to request authorization response parameters (along with additional data) in JWT format instead of plain text.
+
+The specification defines the following response modes, which you can specify using the `response_mode` parameter in the authorization request.
+
+### query.jwt
+  
+Response parameters are encoded in a single JWT and sent as a query parameter to the redirect URI.
+
+Sample response:
+
+```bash
+HTTP/1.1 302 Found
+Location: https://<CLIENT_HOST>/redirects/redirect1?response=<JWT>
+```
+
+### fragment.jwt
+
+Response parameters are encoded in a single JWT and sent as a fragment parameter to the redirect URI.
+
+Sample response:
+
+```bash
+HTTP/1.1 302 Found
+Location: https://<CLIENT_HOST>/redirects/redirect1#response=<JWT>
+```
+
+### form_post.jwt
+
+The response parameter containing the JWT is encoded as an HTML form value that is auto-submitted in the User Agent, and thus is transmitted via the HTTP POST method to the client, with the result parameters being encoded in the body.
+
+Sample response from the authorization server to the user agent:
+
+```bash
+HTTP/1.1 200 OK
+Content-Type: text/html;charset=UTF-8
+Cache-Control: no-cache, no-store
+Pragma: no-cache
+
+<html>
+<head><title>Submit Form</title>
+</head>
+<body onload="javascript:document.forms[0].submit()">
+<form method="post" action="https://<CLIENT_HOST>/redirects/redirect1">
+    <input type="hidden" name="response" value=<JWT>/>
+</form>
+</body>
+</html>
+```
+
+The above response results in the following POST request to the client's redirect URI:
+
+```bash
+POST /redirects/redirect1 HTTP/1.1
+Host: <CLIENT_HOST>
+Content-Type: application/x-www-form-urlencoded
+
+response=<JWT>
+```
+
+### jwt
+
+The response mode jwt is a shortcut and indicates the default redirect encoding (query, fragment) for the requested response type, as shown below.
+
+- If <code>response_type</code> is <code>code</code>, the <code>response_mode</code> is <code>query.jwt</code>.
+- If <code>response_type</code> is <code>token</code>, the <code>response_mode</code> is <code>fragment.jwt</code>.
+- If <code>response_type</code> contains <code>token</code> or <code>id_token</code>, the <code>response_mode</code> is <code>fragment.jwt</code>.
+
```

Comment:
- For the response types defined in [OIDM](https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html) (​​except "none") the default response mode is <code>fragment.jwt</code>
(i.e. for response types containing <code>token</code> or <code>id_token</code> the default response mode is <code>fragment.jwt</code>)